### PR TITLE
Add embedded jdk support for ppc64le

### PR DESCRIPTION
**Pull request by jayfurmanek**
This adds support for embedding OpenJDK in bazel for ppc64le.
Azul Systems does not create Zulu/OpenJDK builds for ppc64le, but AdoptOpenjdk.net does.

This PR adds the necessary configs and conditionals and http_file configuration URLs for adoptopenjdk downloads.
The minimize_jdk.sh script needed a slight generalization.

I've been using adoptOpenjdk-embedded bazel for quite a while and can vouch for its stability.

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -319,6 +319,15 @@ http_file(
     urls = ["https://mirror.bazel.build/openjdk/azul-zulu11.37.48-ca-jdk11.0.6/zulu11.37.48-ca-jdk11.0.6-linux_aarch64-minimal-b23d4e05466f2aa1fdcd72d3d3a8e962206b64bf-1581690750.tar.gz"],
 )
 
+http_file(
+    name = "openjdk_linux_ppc64le_vanilla",
+    downloaded_file_path="adoptopenjdk-ppc64le-vanilla.tar.gz",
+    urls = [
```
Comment (philwo) [inline WORKSPACE:325]: Please add the `sha256` attribute. I mirrored the file now and the SHA of the file is: `a417db0295b1f4b538ecbaf7c774f3a177fab9657a665940170936c0eca4e71a`.

File: src/main/java/com/google/devtools/build/lib/bazel/rules/java/jdk.WORKSPACE [hunk 2]
```diff
@@ -117,6 +117,18 @@ maybe(
     ],
 )
 
+# This must be kept in sync with the top-level WORKSPACE file.
+maybe(
+    http_archive,
+    name = "remotejdk11_linux_ppc64le",
+    build_file = "@local_jdk//:BUILD.bazel",
+    strip_prefix = "jdk-11.0.7+10",
```
Comment (philwo) [inline src/main/java/com/google/devtools/build/lib/bazel/rules/java/jdk.WORKSPACE:125]: Please also add the sha256 attribute here.

File: src/main/java/com/google/devtools/build/lib/bazel/rules/java/jdk.WORKSPACE [hunk 3]
```diff
@@ -117,6 +117,18 @@ maybe(
...
+    urls = [
+        "https://mirror.bazel.com/openjdk/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.2%2B9/OpenJDK11U-jdk_ppc64le_linux_hotspot_11.0.7_10.tar.gz",
+        "https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.2%2B9/OpenJDK11U-jdk_ppc64le_linux_hotspot_11.0.7_10.tar.gz",
```
Comment (philwo) [inline src/main/java/com/google/devtools/build/lib/bazel/rules/java/jdk.WORKSPACE:128]: This URL doesn't seem to be valid and gives an HTTP 404. Can you replace it with the one above?

The difference is one character: This one `jdk-11.0.2%2B9` vs. the above `jdk-11.0.2%2B10`.

**Comment by philwo** [comment]
FYI @meisterT

**Comment by meisterT** [comment]
Nice!

**Comment by jayfurmanek** [comment]
I think the remaining fails here are unrelated.

Also, what's the policy on compile.sh changes?  We add in the below to the bazel build command:

```
--distinct_host_configuration=false \
--javabase=@bazel_tools//tools/jdk:jdk \
```

It looks like it doesn't build in the jdk by default. Is the idea that you should do that kind of thing with a bazelrc file?

Thanks for reviewing!

**Comment by philwo** [comment]
> --distinct_host_configuration=false \
> --javabase=@bazel_tools//tools/jdk:jdk \

To be honest, I'm not sure what these flags do. 😬 Why do you need them for your build?

The test failures are indeed related to this change:

```
ERROR: An error occurred during the fetch of repository 'remotejdk11_linux_ppc64le':
   java.io.IOException: Error downloading [https://mirror.bazel.com/openjdk/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.7%2B10/OpenJDK11U-jdk_ppc64le_linux_hotspot_11.0.7_10.tar.gz, https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.7%2B10/OpenJDK11U-jdk_ppc64le_linux_hotspot_11.0.7_10.tar.gz] to /var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/3385/execroot/io_bazel/_tmp/c5e052945f2150d3644bf8016601089b/root/fc065a228971d6bda964c6ba833756dc/external/remotejdk11_linux_ppc64le/OpenJDK11U-jdk_ppc64le_linux_hotspot_11.0.7_10.tar.gz: Unknown host: github.com
```

They happen because we run most tests without network access on our CI and instead inject the external repositories (like JDKs) from the outside. It should work if you copy the definition of your `remotejdk11_linux_ppc64le` repository into the top-level WORKSPACE file and suffix it with `_for_testing`, similar to this one: https://cs.opensource.google/bazel/bazel/+/master:WORKSPACE;l=674;drc=762c27d2cdc8d66cc4f91a141ec667150e8a5f39

Then add your `remotejdk11_linux_ppc64le_for_testing` repository to these lists similar to the existing ones:
- https://cs.opensource.google/bazel/bazel/+/master:src/test/shell/testenv.sh;l=285;drc=dc602e2a70440e1e8684c360d0b61354eb90ec7c
- https://cs.opensource.google/bazel/bazel/+/master:src/test/py/bazel/test_base.py;l=55;drc=fee6004db115d7f1b974a31299129669be1b18ff
- https://cs.opensource.google/bazel/bazel/+/master:src/BUILD;l=734;drc=cb5dc080dae009c43b30c2981ee7590e8f9c2ea4

That should fix it. :)

**Comment by jayfurmanek** [comment]
That worked. Thanks for the assistance @philwo !

**Comment by jayfurmanek** [comment]
Anything else needed here?  Thx!!

**Comment by philwo** [comment]
@jayfurmanek I think we're good, thanks! I'll try to merge this now.

**Review by blg666 — APPROVED** [review]

**Pull request closed without merging**